### ISSUE TITLE
Issue building new release v1.0.0, Liquid syntax error

### ISSUE THREAD

**Issue by colquittlab** [greenelab/lab-website-template#164]
### Checks

- [X] I have searched **[the docs](https://greene-lab.gitbook.io/lab-website-template-docs)**, [existing issues](https://github.com/greenelab/lab-website-template/issues), and [existing discussions](https://github.com/greenelab/lab-website-template/discussions) for answers first.

### Version of Lab Website Template you are using

1.0.0

### Description

I've been troubleshooting a GitHub Pages build issue but am running up against a wall.

I've tested my site locally and it build correctly. But pushing to GitHub and subsequent build yields this error:

  Liquid Exception: Liquid syntax error (/github/workspace/_includes/head.html line 22): Expected end_of_string but found id included in /_layouts/default.html

I think I have tracked this down to line 22 of _includes/styles.html:

{% assign styles = site.static_files
  | where_exp: "file",
    "file.path contains '/_styles' and file.path contains '.css'"
%}

It looks like 'and' operators are not supported in Jekyll <4.0 and GitHub pages appears to be using v3.9.3.

I edited this Liquid statement to:

{% assign styles = site.static_files
  | where_exp: "file",
    "file.path contains '/_styles'"
    | where_exp: "file",
    "file.path contains '.css'"
%}

But now receive this error:

 Liquid Exception: Invalid scheme format: ' <div class="cols" style="--cols' in /_layouts/default.html

Thoughts? Thank you.

**Comment by vincerubinetti** [comment]
I'm pretty sure I know what is causing this, and it's very dumb and confusing unfortunately.

Try adding a single blank file named `.nojekyll` in the root of your `gh-pages` branch. Also, make sure you've set your repo to build your site from the `gh-pages` branch [as described in the setup instructions](https://greene-lab.gitbook.io/lab-website-template-docs/getting-started/set-up-your-site).

The reason this is needed is because without it, GitHub will attempt to run Jekyll on whatever you have in the repo... even if it's not a Jekyll site! And the version of Jekyll it runs automatically like this is an older, locked down (only [these plugins](https://pages.github.com/versions/) are allowed) version of Jekyll. The template `v1.0.0` doesn't use GitHub's built-in, limited Jekyll. It uses the latest version, as you will see in your `Gemfile`, without any limitations, and builds it "manually" with GitHub Actions (`.github/workflows/build-site.yaml`) any time you push a change to your repo.

So, long story short, without the `.nojekyll` file, GitHub Actions will first build the site with Jekyll 4.3, and then GitHub Pages will try to build the _output of the first build again with an old Jekyll_ that doesn't support that syntax.

Normally [the template does this for you when you set it up](https://github.com/greenelab/lab-website-template/blob/main/.github/workflows/first-time-setup.yaml#L34-L35), but you wouldn't have run that first time set up if you were coming from an old version of the template.

I will add a note about existing users not needing to run the "first time setup" automated process, but then they have to manually do the critical parts of it: create the `gh-pages` branch (if it doesn't exist), and create the `.nojekyll` file.

**Comment by colquittlab** [comment]
Excellent, that fixed it! I had never run the first-time-setup Action (as I thought everything was setup from the previous version...), but running this solved it. Thanks for the help and the excellent template.
